drivers/SmartThings/zwave-garage-door-opener/src/ecolink-zw-gdo/init.lua

```diff
@@ -0,0 +1,458 @@
+
+      elseif (notificationEvent ==
+      Notification.event.access_control.BARRIER_UNABLE_TO_PERFORM_REQUESTED_OPERATION_DUE_TO_UL_REQUIREMENTS) then
+        barrier_event = capabilities.doorControl.door.unknown()
+        log.info_with({hub_logs=true},
+                      string.format("NOTIFICATION ACCESS CONTROL - UNABLE TO PERFORM DUE TO UL"))
+
+      elseif (notificationEvent ==
+      Notification.event.access_control.BARRIER_FAILED_TO_PERFORM_REQUESTED_OPERATION_DEVICE_MALFUNCTION) then
+        barrier_event = capabilities.doorControl.door.unknown()
+        log.info_with({hub_logs=true},
+              string.format("NOTIFICATION ACCESS CONTROL - UNABLE TO PERFORM DUE TO DEVICE MALFUNCTION"))
+
+      elseif (notificationEvent ==
+              Notification.event.access_control.BARRIER_SENSOR_NOT_DETECTED_SUPERVISORY_ERROR) then
+        barrier_event = capabilities.doorControl.door.closed()
+        contact_event = capabilities.healthCheck.healthStatus.online()
+        log.info_with({hub_logs=true},
+                    string.format("NOTIFICATION ACCESS CONTROL - SENSOR NOT DETECTED SUPERVISORY ERROR"))
+
+      elseif (notificationEvent ==
+              Notification.event.access_control.BARRIER_SENSOR_LOW_BATTERY_WARNING) then
+        barrier_event = capabilities.doorControl.door.closed()
+        contact_event = capabilities.battery.battery(CONTACTSENSOR_BATTERY_LEVEL_LOW)
+        log.info_with({hub_logs=true},
+                      string.format("NOTIFICATION ACCESS CONTROL - SENSOR LOW BATTERY WARNING"))
+
+      end
+
+    end
+  end
+
+  -- If we are going to emit an event to the device, from a notification, do it.
+  if (barrier_event ~= nil) then
+    device:emit_event_for_endpoint(GDO_ENDPOINT_NUMBER, barrier_event)
+  end
+
+  if (contact_event ~= nil) then 
+    device:emit_event_for_endpoint(CONTACTSENSOR_ENDPOINT_NUMBER, contact_event)
+  end
+
+end
+
+--- Multilevel Sensor Report Handler
+---
+--- @param driver st.zwave.Driver
+--- @param device st.zwave.Device
+--- @param cmd st.zwave.CommandClass.SensorMultilevel.Report
+local function sensor_multilevel_report_handler(driver, device, cmd)
+  log.info_with({hub_logs=true}, 
+                string.format("sensor_multilevel_report_handler> Type:%d Value:%s", 
+                cmd.args.sensor_type, cmd.args.sensor_value) )
+  -- Handle Temperature Report
+  if (SensorMultilevel.sensor_type.TEMPERATURE == cmd.args.sensor_type) then
+    local scale = 'C'
+    if (SensorMultilevel.scale.temperature.FAHRENHEIT == cmd.args.scale) then
+      scale = 'F'
+    end
+    log.info_with({hub_logs=true}, 
+                  string.format("Temperature> Value:%s Scale:%s", 
+                  tostring(cmd.args.sensor_value), scale) )
+    local event = capabilities.temperatureMeasurement.temperature(
+                                          {value = cmd.args.sensor_value, unit = scale})
+    device:emit_event_for_endpoint(GDO_ENDPOINT_NUMBER, event)
```

Comment:
again, if you've set up your endpoint mapping function correctly, then you shouldn't need to overwrite the defaults here

Reply:
Please elaborate. Without this function the temperature does not update on the smartthings app.

Reply:
I'd verify your endpoint mapping is correct, then. What endpoint do the temperature reports come from?

Reply:
The GDO endpoint

Reply:
You can check this in your tests by specifying which endpoint the commands come from.

here's an example of how to do that: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zwave-switch/src/test/test_zwave_dual_switch.lua#L113